site/docs/relations/logical_relations.md

```diff
@@ -219,6 +219,40 @@ The join operation will combine two separate inputs into a single output, based
     ```
 
 
+## AsOfJoin Operation
+
+The as-of-join operation is a time-series operation that will combine a left input and multiple right inputs into a single output, based on a join expression, an on-field and a tolerance value. All inputs must have the on-field in ascending-order. The operation is similar to a join-operation where the join expression is used for exact matching whereas the on-field is used for inexact matching up to the tolerance value.
```

Comment:
I wonder if the statement "All inputs must have the on-field in ascending-order" is too prescriptive.  In theory, an as-of join could be created even if the on-field wasn't in ascending order by doing something similar to a hash-join and building up a table in memory.  In this case, if the ordering is not present (or not compatible with the on-key) then a consumer that only supports the ordered version can simply reject the plan.

That being said...if no real consumer supports an unordered variant...then maybe this would just be pedantic.

Reply:
The more I think of this the more I think maybe a non-ordered as-of join is just too weird.  For example, distribution would not necessarily be maintained in this case.

Reply:
IIUC, you're not asking here for a change, but let me know if I misunderstood.